src/example/stokes_flow/stokes_flow.jl

```diff
@@ -0,0 +1,354 @@
+# ```math
+# \begin{bmatrix}
+#  A & B^T \\
+#  B & 0 
+# \end{bmatrix} 
+# \begin{bmatrix} 
+# U \\
+# P
+# \end{bmatrix}
+# = L
+# ``` 
+#
+# # Commented code
+
+# import necessary packages
+using Bcube
+using LinearAlgebra
+using WriteVTK
+using StaticArrays
+using SpecialFunctions
+
+# Function space: P2 for velocity and P1 for pressure
+const fspace = :Lagrange
+const degree_u = 2
+const degquad = 2 * degree_u + 1
+const degree_p = 1
+
+# Material parameters
+const ρ = 1400.0
+const μ = 14.0
+const ν = μ / ρ
+
+# Parameters for the oscillating plate case
+const f = 1.0
+const k = sqrt(π * f / ν)
+
+const Δt = 1.0e-3
+const finalTime = 2.0
+
+const outputpath = joinpath(@__DIR__, "..", "..", "..", "myout", "stokes")
+mkpath(outputpath)
+
+# Function that runs the steady case:
+function run_steady()
+    # Read 2D mesh
+    mesh_path =
+        joinpath(@__DIR__, "..", "..", "..", "input", "mesh", "domainTriangle_tri.msh")
+    mesh = read_msh(mesh_path)
+
+    fsu = FunctionSpace(fspace, degree_u)
+    U_vel = TrialFESpace(
+        fsu,
+        mesh,
+        Dict(
+            "edge_right" => SA[0.0, 0.0],
+            "edge_left" => SA[0.0, 0.0],
+            "edge_top" => SA[1.0, 0.0],
+        );
+        size = 2,
+    )
+    V_vel = TestFESpace(U_vel)
+
+    fsp = FunctionSpace(fspace, degree_p)
+    U_pre = TrialFESpace(fsp, mesh, Dict("edge_top" => SA[0.0]); size = 1)
+    V_pre = TestFESpace(U_pre)
+
+    # Define MultiFESpace (mixed formalism)
+    U = MultiFESpace(U_vel, U_pre)
+    V = MultiFESpace(V_vel, V_pre)
+
+    velocity = FEFunction(U_vel, 0.0)
+    pressure = FEFunction(U_pre, 0.0)
+
+    # Define measures for cell
+    dΩ = Measure(CellDomain(mesh), degquad)
+
+    # volume force is equal to zero
+    f = PhysicalFunction(x -> SA[0.0, 0.0])
+
+    # definition of bilinear and linear forms
+    a((u, p), (v, q)) = ∫(μ * ∇(u) ⊡ ∇(v) - tr(∇(v)) * p + tr(∇(u)) * q)dΩ
+    l((v, q)) = ∫(f ⋅ v)dΩ
+
+    # assemble matrix and RHS
+    A = assemble_bilinear(a, U, V)
+    L = assemble_linear(l, V)
+
+    # build Dirichlet vector to apply lift
+    Wd = Bcube.assemble_dirichlet_vector(U, V, mesh)
+
+    # Apply lift
+    L = L - A * Wd
+
+    # Apply homogeneous dirichlet condition
+    Bcube.apply_homogeneous_dirichlet_to_vector!(L, U, V, mesh)
+    Bcube.apply_dirichlet_to_matrix!(A, U, V, mesh)
+
+    # Solve linear system
+    sol = A \ L
+
+    sol = sol .+ Wd
+
+    ϕ = FEFunction(V)
+
+    # Write solution
+    set_dof_values!(ϕ, sol)
+    velocity, pressure = ϕ
+
+    vars_1 = Dict("Velocity" => velocity)
+    Bcube.write_vtk_lagrange(
+        joinpath(outputpath, "output_velocity_steady"),
+        vars_1,
+        mesh,
+        U_vel,
+    )
+    vars_2 = Dict("Pressure" => pressure)
+    Bcube.write_vtk_lagrange(
+        joinpath(outputpath, "output_pressure_steady"),
+        vars_2,
+        mesh,
+        U_pre,
+    )
```

Comment:
could you add a comment to explain why it's better to write results in two seperate vtk files when solutions have different degrees?

Reply:
In fact different degrees weren't necessary for the visualization. I will change this to have a single output file.